src/analysis/timeline/timelineCredView.test.js

```diff
@@ -0,0 +1,109 @@
+    const boundaries = utcWeek.range(startTimeMs, endTimeMs);
+    const intervals = [];
+    for (let i = 0; i < boundaries.length - 1; i++) {
+      intervals.push({
+        startTimeMs: +boundaries[i],
+        endTimeMs: +boundaries[i + 1],
+      });
+    }
+    const users = [
+      ["starter", (x) => Math.max(0, 20 - x)],
+      ["steady", (_) => 4],
+      ["finisher", (x) => (x * x) / 20],
+      ["latecomer", (x) => Math.max(0, x - 20)],
+    ];
+
+    const graph = new Graph();
+    const addressToCred = new Map();
+    for (const [name, generator] of users) {
+      const address = NodeAddress.fromParts(["foo", name]);
+      graph.addNode({
+        address,
+        description: `[@${name}](https://github.com/${name})`,
+        timestampMs: null,
+      });
+      const scores = intervals.map((_unuesd, i) => generator(i));
+      addressToCred.set(address, scores);
+    }
+    const filteredTimelineCred: FilteredTimelineCred = {
+      intervals,
+      addressToCred,
+    };
+    const params = {alpha: 0.05, intervalDecay: 0.5, weights: defaultWeights()};
+    return new TimelineCred(graph, filteredTimelineCred, params, credConfig());
+  }
+
+  it("JSON serialization works", () => {
+    const json = exampleTimelineCred().toJSON();
+    const tc_ = TimelineCred.fromJSON(json, credConfig());
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const tcv_ = new TimelineCredView(tc_);
+    expect(tcv.credSortedNodes(NodeAddress.empty)).toEqual(
+      tcv_.credSortedNodes(NodeAddress.empty)
+    );
+  });
+
+  it("default interval works", () => {
+    const tc = exampleTimelineCred();
+    const tcv = new TimelineCredView(tc);
+    expect(tc.intervals()).toEqual(tcv.intervals());
+  });
+
+  it("single input interval works", () => {
+    const tc = exampleTimelineCred();
+    const startTimeMs = +new Date(2017, 1);
+    const inputInterval = {startTimeMs: startTimeMs, endTimeMs: startTimeMs};
+    const tcv = new TimelineCredView(tc, inputInterval);
+    expect(tcv.intervals().length).toEqual(1);
+  });
+
+  it("straddle input interval works", () => {
+    const tc = exampleTimelineCred();
+    const startTimeMs = +new Date(2017, 0, 14);
+    const endTimeMs = +new Date(2017, 0, 14 + 7 + 2);
+    const inputInterval = {startTimeMs: startTimeMs, endTimeMs: endTimeMs};
+    const tcv = new TimelineCredView(tc, inputInterval);
+    expect(tcv.intervals().length).toEqual(3);
+  });
+
+  it("cred sorting works", () => {
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const sorted = tcv.credSortedNodes(NodeAddress.empty);
+    const expected = sortBy(sorted, (x) => -x.total);
+    expect(sorted).toEqual(expected);
+  });
+
+  it("cred aggregation works", () => {
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const nodes = tcv.credSortedNodes(NodeAddress.empty);
+    for (const node of nodes) {
+      expect(node.total).toEqual(sum(node.cred));
+    }
+  });
```

Comment:
so far, we haven't checked consistency between the intervals and the cred. so it could be that when I request cred for a node, it gives me cred for all of the intervals rather than the filtered ones. or, it could give me cred for the wrong set of intervals, even if it has the right length. we should add a test to ensure the cred and intervals match up properly.